spec/requests/api/v1.0/filter_spec.rb

```diff
@@ -18,6 +18,29 @@ def expect_success(query, *results)
     )
   end
 
+  def expect_failure(collection, query, *errors)
+    get(URI.escape("/api/v1.0/#{collection}?#{query}"))
+
+    expect(response).to(
+      have_attributes(
+        :parsed_body => {
+          "errors" => errors.collect { |e| {"detail" => e, "status" => 400} }
+        },
+        :status      => 400
+      )
+    )
+  end
+
+  def expect_success_ordered_objects(collection, query, results)
+    get(URI.escape("/api/v1.0/#{collection}?#{query}"))
+
+    expect(response.status).to(eq(200))
+
+    response.parsed_body["data"].each do |object|
+      expect(object).to(include(results.shift.deep_stringify_keys))
```

Comment:
Something about this helper feels weird to me.  I would expect it to return an order set that the rest itself can compare with a normal `eq` matcher

Reply:
I had to do something like this here because eq matches the whole hash, but I only cared to match a subset.  For using eq for this purpose, I guess one could do the following:

```
 results = results.collect(&:stringify_keys)
 attrs = results.first.keys

 expect(response.parsed_body["data"].collect { |res| res.slice(*attrs) }).to(eq(results))
```

Reply:
This also allows us to do a single expect vs. one per resource returned.